api/pb/android/threads/src/main/java/io/textile/threads/Client.java

```diff
@@ -1,63 +1,72 @@
 package io.textile.threads;
 
-import java.io.ByteArrayOutputStream;
 import java.util.Arrays;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.concurrent.atomic.AtomicReference;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-
+import java.util.function.Consumer;
 import android.arch.lifecycle.LifecycleObserver;
-
 import com.google.protobuf.ByteString;
-
+import io.grpc.CallCredentials;
 import io.grpc.ManagedChannel;
-import io.grpc.ManagedChannelBuilder;
-
-import java.util.Iterator;
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.TimeUnit;
-
 import io.grpc.stub.StreamObserver;
 import io.textile.threads_grpc.*;
 
-
-class ClientException extends Exception
-{
-    public ClientException(String message)
-    {
-        super(message);
-    }
-}
-
 /**
  * Provides top level access to the Textile API
  */
 public class Client implements LifecycleObserver {
-    private static ManagedChannel channel;
     private static APIGrpc.APIBlockingStub blockingStub;
     private static APIGrpc.APIStub asyncStub;
-    private static String grpcHost;
-    private static int grpcPort;
+    private static Config config;
     enum ClientState {
         Connected, Idle
     }
     public static ClientState state = ClientState.Idle;
 
-    public Client(String host, int port) {
-        grpcHost = host;
-        grpcPort = port;
+    /**
+     * Initialize a new Client
+     */
+    public Client() {
+        this.config = new DefaultConfig();
     }
 
-    public void Connect() {
-        channel = ManagedChannelBuilder
-                .forAddress(grpcHost, Math.toIntExact(grpcPort))
-                .usePlaintext()
-                .build();
-        blockingStub = APIGrpc.newBlockingStub(channel);
-        asyncStub = APIGrpc.newStub(channel);
-        state = ClientState.Connected;
+    /**
+     * Initialize a new Client
+     * @param config is either a DefaultConfig for running threadsd or TextileConfig for using hosted.
+     */
+    public Client(Config config) {
+        this.config = config;
+    }
+
+    /**
+     *
+     * @return the current session id or null
+     */
+    public String getSession() {
+        return this.config.getSession();
+    }
+    /**
+     * Method must be called before using the Client and while the device has an internet connection.
+     * @param ready
+     */
+    public void init(Consumer<Boolean> ready) {
```

Comment:
Doing some reading about `Consumer`s, it's not clear to me that is the correct API here. Seems like it should be returning a `Promise` since there is some async work done in here.

Reply:
what do you mean by promise exactly?

Reply:
Sorry, meant `Future` https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html